Ba/10 journal management improvements

beat2319 opened the pull request:
## Feat: Journal Improvements (Delete, Dynamic Textarea, Nav Fix) & API Tests

This PR introduces several updates to the Journal feature and testing:

### Key Changes:

* **Journal Deletion:**
    * Added delete button/form to `journal-card.hbs`.
    * Implemented backend route `POST /api/journal/:id` with ownership check and redirects.
* **Dynamic Textarea:**
    * The `#journal-entry` textarea in the modal now auto-resizes with content using JavaScript (`scrollHeight`).
    * Applied necessary Tailwind classes (`resize-none`, `max-h`, `min-h`, `overflow-hidden`) for styling and bounds.
* **UI Fix:**
    * Prevented nav sidebar shrinking by adding `flex-shrink-0` to the `<aside>` element.
* **Journal API Tests (`server.spec.js`):**
    * Added Mocha/Chai-Http test suite for Journal API endpoints (`GET /journal`, `POST /api/journal`, `POST /api/journal/:id`).
    * Uses `chai.request.agent(app)` for authenticated requests (fixing previous hook issues).
    * Covers successful add/delete, validation errors, and not found cases.
    * Includes debug fixes (using `app` directly, `.redirects(0)`, correct status/header checks for redirects).

### Testing:

* [x] Added backend API tests for Journal endpoints.
* [x] Manually verified UI changes (delete button, textarea resizing, nav fix).

Ben-Mast commented on ProjectSourceCode/js/app.js:
```diff
@@ -223,34 +223,55 @@ if (window.location.pathname === "/settings/pal-settings") {
   });
 }
 
-//ensures search message stays after friends list reload
-window.addEventListener("DOMContentLoaded", () => {
-  const storedMessage = localStorage.getItem("searchMessage");
-  if (storedMessage) {
-      document.getElementById("search-result").textContent = storedMessage;
-      localStorage.removeItem("searchMessage"); // Clear it after displaying
-  }
-});
```
Don't delete this, it breaks Sam's message system.

Ben-Mast commented on ProjectSourceCode/js/app.js:
```diff
@@ -223,34 +223,55 @@ if (window.location.pathname === "/settings/pal-settings") {
-  if (journalAddButton) {
-    journalAddButton.addEventListener("click", function () {
+  // --- Textarea Auto-Resizing Logic ---
+  const autoResizeTextarea = () => {
```
I'm not really a fan of the auto-resize on the modal. It makes it so that if you type a long message, the top and bottom of the modal expand so much that they go off the screen, and the submit button becomes inaccessible. I think just having a fixed sized journal body that scrolls is easier and more accessible for the user.

Ben-Mast replied:
Actually it works fine with the older modal, it's just because of the narrower new modal that this happens.

Ben-Mast commented on ProjectSourceCode/views/partials/add-journal-modal.hbs:
```diff
@@ -11,28 +11,73 @@
     <p>Journal Entry</p>
 </button>
 
-<div class="z-50 fixed inset-0 items-center justify-center bg-black/50 hidden" id="journal-modal">
-    <form action="/api/journal" method="POST" id="journal-form"
-        class="bg-white rounded-lg shadow-lg p-6 w-1/2 max-w-128 mx-auto mt-20">
-        <div class="flex justify-between mb-4">
-            <h2 class="text-2xl font-bold mb-4">Journal Entry</h2>
-            <svg xmlns="http://www.w3.org/2000/svg" id="journal-close-modal-button" width="24" height="24" viewBox="0 0 24 24"
-                fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"
-                class="cursor-pointer icon icon-tabler icons-tabler-outline icon-tabler-x">
-                <path stroke="none" d="M0 0h24v24H0z" fill="none" />
-                <path d="M18 6l-12 12" />
-                <path d="M6 6l12 12" />
-            </svg>
-        </div>
-        <label for="journal-title" class="block text-sm font-medium text-gray-700 mb-2">Title</label>
-        <input type="text" id="journal-title" name="journal-title"
-            class="w-full p-2 border border-gray-300 rounded mb-4" required maxlength="100">
-        <label for="journal-entry" class="block text-sm font-medium text-gray-700 mb-2">Entry</label>
-        <textarea id="journal-entry" name="journal-entry"
-            class="w-full p-2 border border-gray-300 rounded mb-4" maxlength="1000"></textarea>
-        <button type="submit" class="bg-blue-400 hover:bg-blue-500 text-white px-4 py-2 rounded cursor-pointer">Add
-            Entry</button>
-        <button type="button" id="journal-cancel-button"
-            class="bg-gray-300 text-black px-4 py-2 rounded ml-2 cursor-pointer">Cancel</button>
-    </form>
```
I actually think this older modal was better design. The newer one has a very odd cramped width. It's the cramped width + the auto-resize that gets really unfortunate with large inputs.

Ben-Mast reviewed (changes requested):
The journal deletion and testing are great! Some of the new UI changes introduce more issues than they solve I think.

There is one important comment I left on app.js because a message function got deleted accidently in a merge conflict. Once that's fixed all of the UI stuff is subjective so that's up to you if you wanna change it or leave it as is, all good either way.

beat2319 commented:
Just fixed so stuff, then broke it badly(got to excited with a new feature), then reverted to an old commit and updated some unneeded logic and modal issues. Should be working!!

Ben-Mast reviewed (approved):
Looks good! You removed the frontend limit on the journal entry so you might wanna add the maxLength attribute back to that instead of relying only on the backend.

beat2319 commented:
yes ill do that right now, I wanted to switch back to TEXT to allow paragraphs. anyway it didnt work as well as I had hoped, so yes that will be added back

anamanica commented:
This looks like some gooooood stuff. I just pulled the repo onto my machine locally, and everything seems to be working beautifully! I like that you added some test cases!
